sdks/java/extensions/ordered/src/main/java/org/apache/beam/sdk/extensions/ordered/OrderedProcessingStatus.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.ordered;
+
+import com.google.auto.value.AutoValue;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+import org.joda.time.Instant;
+
+/** Indicates the status of ordered processing for a particular key. */
+@AutoValue
+@DefaultSchema(AutoValueSchema.class)
+public abstract class OrderedProcessingStatus {
+
+  public static OrderedProcessingStatus create(
+      Long lastOutputSequence,
+      long numberOfBufferedEvents,
+      Long earliestBufferedSequence,
+      Long latestBufferedSequence,
+      long numberOfReceivedEvents,
+      long resultCount,
+      long duplicateCount,
+      boolean lastEventReceived) {
+    return new AutoValue_OrderedProcessingStatus.Builder()
+        .setLastProcessedSequence(lastOutputSequence)
+        .setNumberOfBufferedEvents(numberOfBufferedEvents)
+        .setEarliestBufferedSequence(earliestBufferedSequence)
+        .setLatestBufferedSequence(latestBufferedSequence)
+        .setNumberOfReceivedEvents(numberOfReceivedEvents)
+        .setLastEventReceived(lastEventReceived)
+        .setDuplicateCount(duplicateCount)
+        .setResultCount(resultCount)
+        .setStatusDate(Instant.now())
+        .build();
+  }
+
+  @Nullable
+  public abstract Long getLastProcessedSequence();
```

Comment:
Similarly, these functions could probably use comments explaining how they fit in for someone who wants to use the output downstream

Reply:
Sorry, missed this class. Added JavaDocs.